library/coretests/tests/simd_intrinsics.rs

```diff
@@ -0,0 +1,1483 @@
+#![allow(non_camel_case_types)]
+
+use core::cmp::{max, min};
+use core::intrinsics::simd::*;
+
+#[repr(simd, packed)]
+#[derive(Copy, Eq)]
+pub struct Simd<T, const N: usize>(pub [T; N]);
+
+impl<T: Copy, const N: usize> Clone for Simd<T, N> {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+impl<T: [const] PartialEq, const N: usize> const PartialEq for Simd<T, N> {
+    fn eq(&self, other: &Self) -> bool {
+        self.as_array() == other.as_array()
+    }
+}
+
+impl<T: core::fmt::Debug, const N: usize> core::fmt::Debug for Simd<T, N> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> Result<(), core::fmt::Error> {
+        <[T; N] as core::fmt::Debug>::fmt(self.as_array(), f)
+    }
+}
+
+impl<T, const N: usize> const core::ops::Index<usize> for Simd<T, N> {
+    type Output = T;
+    fn index(&self, i: usize) -> &T {
+        &self.as_array()[i]
+    }
+}
+
+impl<T, const N: usize> Simd<T, N> {
+    pub const fn from_array(a: [T; N]) -> Self {
+        Simd(a)
+    }
+
+    pub const fn splat(value: T) -> Self
+    where
+        T: Copy,
+    {
+        Self([value; N])
+    }
```

Comment:
you need to use `mem::transmute` instead of accessing the field inside a `repr(simd)` struct, since last I knew the compiler doesn't properly handle the field accesses in all cases

```suggestion
    pub const fn from_array(a: [T; N]) -> Self {
        mem::transmute(a)
    }

    pub const fn splat(value: T) -> Self
    where
        T: Copy,
    {
        mem::transmute([value; N])
    }
```